Infoj Order Correction

RobAndrewHurst opened the pull request:
## Description

The lookup for the infoj_order checks on key, then field, and finally query. This fails when a key exists but doesn't match.

We have created a set of the different fields that will then be compared with a has check on the entry.

## [GitHub Issue](https://github.com/GEOLYTIX/xyz/issues/1497)

## Type of Change
Please delete options that are not relevant, and select all options that apply. 

- ✅ Bug fix (non-breaking change which fixes an issue)
- ✅ Testing

## How have you tested this? 
Run the local tests.
## Testing Checklist 

- ✅ Existing Tests still pass
- ✅ New Tests Added
- ✅ Ran locally on my machine

dbauszus-glx reviewed (approved):
Looking good.

I moved the map method into documented function to make this eassier readable.

![image](https://github.com/user-attachments/assets/ecf3e0ac-9a38-4591-9a46-64bfcc5f5efa)

I updated the test to be in order of infoj > order > execution > results > expected > test for readability especially when setting breakpoints.

![image](https://github.com/user-attachments/assets/a3fb55b0-2d8a-4a60-a7f6-1f7b160f8abe)

Note that you can create an array from an elements list allowing to use map functions instead of pushing into results in a for loop.

```js
      const results = Array.from(infoj.children)
        .map(el => el.firstChild.innerText.trim())
```

AlexanderGeere reviewed (approved):
lgtm

RobAndrewHurst commented:
> Looking good.
> 
> I moved the map method into documented function to make this eassier readable.
> 
> ![image](https://private-user-images.githubusercontent.com/22201617/371463895-ecf3e0ac-9a38-4591-9a46-64bfcc5f5efa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc0Mjk2MzYsIm5iZiI6MTcyNzQyOTMzNiwicGF0aCI6Ii8yMjIwMTYxNy8zNzE0NjM4OTUtZWNmM2UwYWMtOWEzOC00NTkxLTlhNDYtNjRiZmNjNWY1ZWZhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI3VDA5Mjg1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA0YzNhNmIyOGFiZGExMWVmZTFhMWI4ZDM0NTQ5Njc1NTEzZjBiODQxNmIwMDU4NTM2ZGZmNWFlZmQ5YWM3ZDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.hzKYZV6f_UOVHLQDFm0zD03OcQkLy2bjI0FMgYK90PU)
> 
> I updated the test to be in order of infoj > order > execution > results > expected > test for readability especially when setting breakpoints.
> 
> ![image](https://private-user-images.githubusercontent.com/22201617/371465032-a3fb55b0-2d8a-4a60-a7f6-1f7b160f8abe.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc0Mjk2MzYsIm5iZiI6MTcyNzQyOTMzNiwicGF0aCI6Ii8yMjIwMTYxNy8zNzE0NjUwMzItYTNmYjU1YjAtMmQ4YS00YTYwLWE3ZjYtMWY3YjE2MGY4YWJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI3VDA5Mjg1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVhMjk3YmQ3NzdmMWQ3MzAyZTRhM2RiNzk2ZGNjYjZjMTFmMWZiZmJlYjIyZDZmYzk2MTc3ZmFlZTBhMjYxNjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.zHlQTsPdtjj4UqtLW5uPSznoYc2MIAK9Aeh2nDgW5y8)
> 
> Note that you can create an array from an elements list allowing to use map functions instead of pushing into results in a for loop.
> 
> ```js
>       const results = Array.from(infoj.children)
>         .map(el => el.firstChild.innerText.trim())
> ```

Nice one!
